testsuite/tests/typing-gadts/pr7269.ml

```diff
@@ -11,15 +11,15 @@ let _ = f (T (`Conj `B) :> s t);; (* warn *)
 type s = [ `A | `B ]
 and sub = [ `B ]
 type +'a t = T : [< `Conj of 'a & sub | `Other of string ] -> 'a t
-Line 4, characters 6-47:
+Line 4, characters 6-28:
 4 | let f (T (`Other msg) : s t) = print_string msg;;
-          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+          ^^^^^^^^^^^^^^^^^^^^^^
 Warning 8 [partial-match]: this pattern-matching is not exhaustive.
 Here is an example of a case that is not matched:
 T (`Conj _)
 
 val f : s t -> unit = <fun>
-Exception: Match_failure ("", 4, 6).
+Exception: Match_failure ("", 4, 7).
```

Comment:
What's happening here: previously the location for `Match_failure`s was based on the start of the `Pexp_fun` node (which for patterns `(A : t)` would be the position of `(`); now the location for `Match_failure`s is based on the start of the pattern stored in the typedtree (which for patterns `(A : t)` is the position of `A`, as type constraints are moved into `tpat_extra` in typedtree).

[The third component of `Match_failure` is documented as the column number where the `match` keyword appears in the source code.](https://v2.ocaml.org/api/Stdlib.html)

If we wanted to raise an exception with a backward-compatible column number here, we'd have to store another location in typedtree (that of the parameter as a whole). It doesn't seem hard, but it would involve more locations in typedtree. WDYT?

Reply:
Outside of this PR conversation, I'm reaching out to merlin developers to make sure they're aware that this PR changes the way that locations work for function arguments. (This test changing output is a concrete example of that, but speaking more generally, `Texp_function` looks rather different than it used to.) As you said, the fix is easy, so I'm not worried about moving ahead for now and revisiting this if merlin developers think this is problematic.